src/org/traccar/protocol/EelinkProtocolDecoder.java

```diff
@@ -97,28 +114,72 @@ private String decodeAlarm(Short value) {
     }
 
     private void decodeStatus(Position position, int status) {
+
+        // GPS - 0 bit
+        position.setValid(BitUtil.check(status, 0));
+
+        // ACC - 1,2 bits
         if (BitUtil.check(status, 1)) {
             position.set(Position.KEY_IGNITION, BitUtil.check(status, 2));
         }
+
+        // Alarm - 3,4 bits
         if (BitUtil.check(status, 3)) {
             position.set(Position.KEY_ARMED, BitUtil.check(status, 4));
         }
+
+        // Remote Cutout - 5,6 bits
         if (BitUtil.check(status, 5)) {
             position.set(Position.KEY_BLOCKED, !BitUtil.check(status, 6));
         }
+
+        // Charge - 7,8 bits
         if (BitUtil.check(status, 7)) {
             position.set(Position.KEY_CHARGE, BitUtil.check(status, 8));
         }
+
+        // Reserved - 9,10,11 bits
+
+        // Digital input 1 - 12 bit - def 0 low
+        position.set(Position.PREFIX_IN + 1, BitUtil.check(status, 12));
+
+        // Digital input 2 - 13 bit - def 1 high
+        position.set(Position.PREFIX_IN + 2, BitUtil.check(status, 13));
+
+        // Digital input 3 - 14 bit - def 0 low
+        position.set(Position.PREFIX_IN + 3, BitUtil.check(status, 14));
+
+        // Digital input 4 - 15 bit - def 1 high
+        position.set(Position.PREFIX_IN + 4, BitUtil.check(status, 15));
+
+        // log the whole status packet
         position.set(Position.KEY_STATUS, status);
     }
 
-    private Position decodeOld(DeviceSession deviceSession, ChannelBuffer buf, int type, int index) {
+    private Position decodeDownlink(
+            DeviceSession deviceSession, ChannelBuffer buf, Channel channel, Position position, int type, int index) {
+
+        getLastLocation(position, new Date());
+        position.setValid(false);
```

Comment:
Downlink data does not come with GPS data, it could be delayed response for some reason and the last location could be incorrect. This was in part why I was asking if there was a way to log a message without Position.